src/ui_common.c

```diff
@@ -1,18 +1,95 @@
 #include "ui.h"
+#include "ux.h"
 
 #include "globals.h"
 #include "os.h"
 
-void io_seproxyhal_display(const bagl_element_t *element);
-
+#ifdef HAVE_BAGL
 void io_seproxyhal_display(const bagl_element_t *element) {
     return io_seproxyhal_display_default((bagl_element_t *)element);
 }
+#endif // HAVE_BAGL
 
 void ui_init(void) {
     UX_INIT();
 }
 
+unsigned char io_event(__attribute__((unused)) unsigned char channel) {
+    // nothing done with the event, throw an error on the transport layer if
+    // needed
+
+    // can't have more than one tag in the reply, not supported yet.
+    switch (G_io_seproxyhal_spi_buffer[0]) {
+    case SEPROXYHAL_TAG_FINGER_EVENT:
+        UX_FINGER_EVENT(G_io_seproxyhal_spi_buffer);
```

Comment:
Could be put inside `#ifdef HAVE_NBGL` guards

Reply:
This even predates Stax / NBGL, so not necessary.